crab/actions/visual_prompt_actions.py

```diff
@@ -0,0 +1,336 @@
+            center_close = ((center_i[0] - center_j[0]) ** 2 + 
+                            (center_i[1] - center_j[1]) ** 2 < 
+                            center_dis_thresh**2)
+            # fmt: on
+            if center_close:
+                boxes_to_remove.add(j)
+
+    boxes_filt = [
+        box for idx, box in enumerate(boxes_with_label) if idx not in boxes_to_remove
+    ]
+    return boxes_filt
+
+
+def _box_a_in_b(a: BoxType, b: BoxType):
+    return a[0] > b[0] and a[1] > b[1] and a[2] < b[2] and a[3] < b[3]
+
+
+def _filter_boxes_by_overlap(boxes_with_label):
+    boxes = [box[0] for box in boxes_with_label]
+    boxes_to_remove = set()
+    for i in range(len(boxes)):
+        if i in boxes_to_remove:
+            continue
+        for j in range(len(boxes)):
+            if _box_a_in_b(boxes[i], boxes[j]):
+                boxes_to_remove.add(j)
+
+    boxes_filt = [
+        box for idx, box in enumerate(boxes_with_label) if idx not in boxes_to_remove
+    ]
+    return boxes_filt
+
+
+def _filter_boxes_by_iou(
+    boxes_with_label: list[tuple[BoxType, str]], iou_threshold=0.5
+):
+    boxes = [box[0] for box in boxes_with_label]
+    boxes_to_remove = set()
+    for i in range(len(boxes)):
+        if i in boxes_to_remove:
+            continue
+        for j in range(i + 1, len(boxes)):
+            iou = _calculate_iou(boxes[i], boxes[j])
+            if iou >= iou_threshold:
+                boxes_to_remove.add(j)
+
+    boxes_filt = [
+        box for idx, box in enumerate(boxes_with_label) if idx not in boxes_to_remove
+    ]
+    return boxes_filt
+
+
+def _draw_boxes(
+    image: Image.Image,
+    boxes: list[BoxType],
+    font_size: int = 30,
+) -> None:
+    draw = ImageDraw.Draw(image)
+    for idx, box in enumerate(boxes):
+        color = tuple(np.random.randint(64, 191, size=3).tolist())
+        font = ImageFont.load_default(font_size)
+        center = _calculate_center(box)
+
+        draw.rectangle([box[0], box[1], box[2], box[3]], outline=color, width=2)
+
+        if hasattr(font, "getbbox"):
+            _, _, w, h = draw.textbbox((0, 0), str(idx), font)
+        else:
+            w, h = draw.textsize(str(idx), font)
+        if box[0] >= w:
+            bbox = (
+                round(box[0] - w),
+                round(center[1] - h / 2),
+                round(box[0]),
+                round(center[1] + h / 2),
+            )
+        else:
+            bbox = (
+                round(box[2]),
+                round(center[1] - h / 2),
+                round(box[2] + w),
+                round(center[1] + h / 2),
+            )
+
+        draw.rectangle(bbox, fill=color)
+        draw.text((bbox[0], bbox[1]), str(idx), fill="white", font=font)
+
+
+@cache
+def _get_grounding_dino_model(type="tiny"):
+    """Get the grounding dino model.
+
+    Args:
+        type (str, optional): "tiny" or "base". Defaults to "tiny".
+
+    Returns:
+        tuple[processer, model]: Tuple of processor and model.
```

Comment:
```suggestion
        tuple[processor, model]: Tuple of processor and model.
```
Also maybe it's better to use class names instead of `processor` and `model` here.

Reply:
Change to the google style docstring.